src/js/components/AppListItemLabelsComponent.jsx

```diff
@@ -0,0 +1,159 @@
+var classNames = require("classnames");
+var React = require("react/addons");
+
+var OnClickOutsideMixin = require("react-onclickoutside");
+var Util = require("../helpers/Util");
+
+var AppListItemLabelsComponent = React.createClass({
+  displayName: "AppListItemLabelsComponent",
+
+  initialMarginTop: null,
```

Comment:
I hear you, and of course I'm aware this is not a best practice. Problem is, you can't setState after the component has been updated (as is the case here, `recalculateDropdownPosition` necessarily has to be called from `componentDidUpdate` to calculate its measurements after rendering).
Also, this variable is supposed to hold the "normal" **initial** topMargin of the dropdown element, and we need to keep track of this value "globally" (on a per-component base). If this was a local variable in the function, the value would be lost at each render pass. 
Another approach would be to use the data-attribute on the element itself. Do you think that's a better idea?